tests/ChannelsEndpointTest.php

```diff
@@ -0,0 +1,308 @@
+
+class ChannelEndpointsTest extends TestCase
+{
+    public function testThatNothingWasReturnOnGetAllChannels()
+    {
+        $response = $this->call('GET', '/api/v1/channels');
+        $output = json_decode($response->getContent());
+
+        $this->assertEquals($output->message, 'Channels are not available for display');
+    }
+
+    public function testGetAllChannels()
+    {
+        $user = factory('Suyabay\User')->create();
+        $channel = factory('Suyabay\Channel', 5)->create();
+
+        $response = $this->call('GET', '/api/v1/channels');
+        $channels = json_decode($response->getContent());
+
+        $channels->data[0] = (array) $channels->data[0];
+        
+        $this->assertTrue(is_array($channels->data[0]));
+        $this->assertArrayHasKey('channel_id', $channels->data[0]);
+        $this->assertArrayHasKey('channel_note', $channels->data[0]);
+        $this->assertArrayHasKey('channel_name', $channels->data[0]);
+        
+    }
+
+    public function testGetASingleChannel()
+    {
+        $channel = factory('Suyabay\Channel')->create([
+            'channel_name' => 'Ginger',
+            'channel_description' => 'Laoriosam volup atum nesciunt beatae dolorem neque ut fuga est',
+            'user_id' => 1,
+        ]);
+        
+        $channel = ChannelRepository::find($channel['id'])->toArray();
+
+        $this->assertTrue(is_array($channel));
+        $this->assertArrayHasKey('user_id', $channel);
+        $this->assertArrayHasKey('channel_description', $channel);
+        $this->assertArrayHasKey('channel_name', $channel);
+
+        $this->get('/api/v1/channels/Ginger')
+        ->seeJson()
+        ->seeStatusCode(200);
+    }
+
+    public function testThatASingleChannelWasNotFound()
+    {
+        $channel = factory('Suyabay\Channel')->create([
+            'channel_name' => 'Suyabay',
+            'channel_description' => 'Laoriosam volup atum nesciunt',
+            'user_id' => 1,
+        ]);
+
+        $response = $this->call('GET', '/api/v1/channels/Suyabaye');
+        $output = json_decode($response->getContent());
+
+        $this->assertEquals(count($output->data), 0);
+    }
+
+    public function testThatAChannelWasCreated()
+    {
+        $user  = $channel = factory('Suyabay\User')->create();
+
+        $channel = factory('Suyabay\Channel')->create([
+            'channel_name' => 'Cabinet',
+            'channel_description' => 'Laoriosam volup atum nesciunt',
+            'user_id' => $user->id,
+        ]);
+
+        $response = $this->call('POST', '/api/v1/channels',
+            [
+                'name' => 'Nyama Choma',
+                'description' => 'This is kenyan way of naming roasted meat',
+            ]
+        );
+
+        $decodedResponse = json_decode($response->getContent());
+
+        $this->assertEquals($decodedResponse->message, 'Channel created successfully');
+    }
+
+    public function testThatOnlyChannelNameOnlyWasSupplied()
```

Comment:
**Code Quality - Nitpick**
@andela-tolotin Any test that ends up having the same error message, should be merge into one test, to help keep the Code Dry